packages/backend/src/api/user/userService.ts

```diff
@@ -25,7 +25,7 @@ export const userService = {
         return new ServiceResponse(ResponseStatus.Success, 'No Users found', null, StatusCodes.NOT_FOUND);
       }
       logger.info('Users retuned successfully');
-      return new ServiceResponse<User[]>(ResponseStatus.Success, 'Users found', users, StatusCodes.OK);
+      return new ServiceResponse(ResponseStatus.Success, 'Users found', users as User[], StatusCodes.OK);
```

Comment:
why change from one to the other? does the other way have any benefits? the first option seems readable to me

Reply:
In the first option, users type is userResponse (as returned from the repository layer) which is incompatible with User[] (different types) The type checking was giving warning. The functionality works fine though. The second implementation takes the userResponse users as a User and the type checking system stopped complaining.
Do you want me to refactor it so that the Service layer functions return a ServiceResponse<UserResponse> instead of ServiceResponse<User> and go back to the old way?

Reply:
or do you want the Repository layer to return UserSchema type (User) instead of userResponse?